modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/FileWriteAheadLogManager.java

```diff
@@ -482,9 +487,30 @@ public void setFileIOFactory(FileIOFactory ioFactory) {
 
             serializer = new RecordSerializerFactoryImpl(cctx).createSerializer(serializerVer);
 
-            GridCacheDatabaseSharedManager dbMgr = (GridCacheDatabaseSharedManager)cctx.database();
+            IgniteCacheDatabaseSharedManager dbMgr = cctx.database();
 
-            metrics = dbMgr.persistentStoreMetricsImpl();
+            if (dbMgr instanceof GridCacheDatabaseSharedManager)
+                metrics = ((GridCacheDatabaseSharedManager)dbMgr).persistentStoreMetricsImpl();
+            else {
+                DataStorageConfiguration dsCfg = cctx.gridConfig().getDataStorageConfiguration();
```

Comment:
I think we should to move this logic up to `IgniteCacheDatabaseSharedManager` by analogue with the `#startMemoryRestore()` method. WDYT?